pyartifactory/objects/build.py

```diff
@@ -0,0 +1,191 @@
+                    create_build_request.number,
+                    create_build_request.name,
+                )
+            except requests.exceptions.HTTPError as error:
+                self._raise_exception(error)
+        else:
+            logger.error("Build %s in %s already exists", create_build_request.number, create_build_request.name)
+            raise ArtifactoryError(f"Build {create_build_request.number} in {create_build_request.name} already exists")
+
+    def promote_build(
+        self,
+        build_name: str,
+        build_number: str,
+        promotion_request: BuildPromotionRequest,
+    ) -> BuildPromotionResult:
+        """
+        :param build_name: Build name to be promoted
+        :param build_number: Build number to be promoted
+        :param promotion_request: Model object containing parameters for promotion
+        :return: BuildPromotionResponse containing server response
+        """
+        try:
+            self._get(
+                f"api/{self._uri}/{build_name}/{build_number}",
+            )
+        except requests.exceptions.HTTPError as error:
+            self._raise_exception(error)
+        else:
+            try:
+                response = self._post(
+                    f"api/{self._uri}/promote/{build_name}/{build_number}",
+                    json=promotion_request.model_dump(),
+                )
+                logging.debug(
+                    "Build %s in %s promoted from %s to %s",
+                    build_number,
+                    build_name,
+                    promotion_request.sourceRepo,
+                    promotion_request.targetRepo,
+                )
+            except requests.exceptions.HTTPError as error:
+                self._raise_exception(error)
+
+        return BuildPromotionResult(**response.json())
+
+    def list(self) -> BuildListResponse:
+        """
+        :return: BuildListResponse model object containing server response
+        """
+        response = self._get(f"api/{self._uri}")
+        logger.debug("List all builds successful")
+        build_list: BuildListResponse = BuildListResponse.model_validate(response.json())
+        return build_list
+
+    def delete(self, delete_build: BuildDeleteRequest) -> None:
+        """
+        :param delete_build: Model object containing required parameters
+        :return: None
+        """
+        try:
+            for _build_number in delete_build.buildNumbers:
+                self._get(
+                    f"api/{self._uri}/{delete_build.buildName}/{_build_number}",
+                )
+        except requests.exceptions.HTTPError as error:
+            self._raise_exception(error)
+        else:
+            # all build numbers exist
+            _del = self._post(f"api/{self._uri}/delete", json=delete_build.model_dump())
+            logger.debug("Builds %s deleted from %s", ",".join(delete_build.buildNumbers), delete_build.buildName)
```

Comment:
Why not just:
```
try:
            for _build_number in delete_build.buildNumbers:
                self._get(
                    f"api/{self._uri}/{delete_build.buildName}/{_build_number}",
                )
            self._post(f"api/{self._uri}/delete", json=delete_build.model_dump())
            logger.debug("Builds %s deleted from %s", ",".join(delete_build.buildNumbers), delete_build.buildName)
        except requests.exceptions.HTTPError as error:
            self._raise_exception(error)
```

This is simpler in my opinion. Applies to all try-except, else blocks. You need to make sure that this is actually necessary before using it